### PR TITLE
feat: Add Nervos DAO generation support

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: packages/common-scripts/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
-import { TransactionSkeleton } from "@ckb-lumos/types"
+import { TransactionSkeleton } from "@ckb-lumos/helpers"
 
 export type Address = string
 export type Config = any // TODO: define this type later
+export type FullCell = any
```
Comment (xxuejie) [inline packages/common-scripts/index.d.ts:5]: Let's take some time to define FullCell here, Config might need more thinking, hence it is a pending item, but FullCell can be defined now.

File: packages/common-scripts/index.d.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+export type CellProvider = any
```
Comment (xxuejie) [inline packages/common-scripts/index.d.ts:6]: `CellProvider` can also be defined now:

1. Let's define a `QueryOptions` structure that might take lock, type, typeIsNull and data.
2. A `CellProvider` is just one type with a `collector` method that takes a `QueryOptions`, and emits a [CellCollector](https://github.com/xxuejie/ckb-js-toolkit/blob/master/index.d.ts#L132)

File: packages/common-scripts/index.d.ts (hunk 1 above)
Comment (classicalliu) [reply packages/common-scripts/index.d.ts]: updated

File: packages/common-scripts/index.d.ts (hunk 2 above)
Comment (classicalliu) [reply packages/common-scripts/index.d.ts]: updated

File: packages/common-scripts/index.d.ts [hunk 3]
```diff
@@ -86,4 +168,114 @@ export declare const secp256k1Blake160Multisig: {
...
+      config: Config,
+    },
+  ): Promise<TransactionSkeleton>,
+
+  /**
+   * list DAO cells,
+   *
+   * @param cellProvider
+   * @param fromAddress
+   * @param cellType "deposit" or "withdraw", can using "" to list all.
```
Comment (xxuejie) [inline packages/common-scripts/index.d.ts:232]: Whenever you need comment like this, please create a new type with the 2 possible values.

File: packages/common-scripts/lib/secp256k1_blake160.js [hunk 4]
```diff
@@ -158,6 +160,7 @@ async function transfer(
       ) {
         continue;
       }
+
```
Comment (xxuejie) [inline packages/common-scripts/lib/secp256k1_blake160.js:163]: Please clean up changes like this.

File: packages/types/index.d.ts [hunk 5]
```diff
@@ -1 +1,63 @@
...
+    type?: Script;
+  };
+  data: string;
+  out_point: OutPoint;
+  block_hash: string;
+  block_number: string;
+}
+
+export interface QueryOptions {
+  lock?: Script | null;
```
Comment (xxuejie) [inline packages/types/index.d.ts:32]: I think we can just allow `Script` or `undefined`, and reject `null`

File: packages/common-scripts/index.d.ts (hunk 3 above)
Comment (classicalliu) [reply packages/common-scripts/index.d.ts]: OK, updated now.

File: packages/common-scripts/lib/secp256k1_blake160.js (hunk 4 above)
Comment (classicalliu) [reply packages/common-scripts/lib/secp256k1_blake160.js]: updated.

File: packages/types/index.d.ts (hunk 5 above)
Comment (classicalliu) [reply packages/types/index.d.ts]: updated

**Pull request merged**